From b9eb9b89ee0bf4aa742faae4b3d28db164a34f87 Mon Sep 17 00:00:00 2001
From: Chen Jiahao <chenjiahao16@huawei.com>
Date: Sat, 28 May 2022 15:52:59 +0800
Subject: [PATCH 2184/2592] [Huawei] KVM: SEV: fix NULL
 kvm_x86_ops.guest_memory_reclaimed calling problem

Offering: HULK
hulk inclusion
category: bugfix
bugzilla: 186806

--------------------------------

commit 8e51c134c3b1 ("KVM: SEV: add cache flush to solve SEV cache
incoherency issues") was backported to mitigate CVE-2022-0171, which
introduced a function pointer kvm_x86_ops.guest_memory_reclaimed.
However the function pointer only defined in svm/sev.c when
CONFIG_KVM_AMD is set.

This commit fix the underlying error when calling NULL pointer
kvm_x86_ops.guest_memory_reclaimed(kvm) in non-AMD platform.

Fixes: 8e51c134c3b1 ("[Backport] KVM: SEV: add cache flush to solve SEV cache incoherency issues")
Signed-off-by: Chen Jiahao <chenjiahao16@huawei.com>
---
 arch/x86/kvm/x86.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/arch/x86/kvm/x86.c b/arch/x86/kvm/x86.c
index 8152c7464641..3c750f7e7983 100644
--- a/arch/x86/kvm/x86.c
+++ b/arch/x86/kvm/x86.c
@@ -8897,6 +8897,9 @@ void kvm_arch_mmu_notifier_invalidate_range(struct kvm *kvm,
 
 void kvm_arch_guest_memory_reclaimed(struct kvm *kvm)
 {
+	if (!kvm_x86_ops.guest_memory_reclaimed)
+		return;
+
 	kvm_x86_ops.guest_memory_reclaimed(kvm);
 }
 
-- 
2.17.1

